### PR TITLE
Sheets Endpoint

### LINKED ISSUE #29: [DEV] Sheets API Endpoint
# Development Task 🔧
## What? ❓
Create a POST /user/sheets endpoint

This will allow somebody to append a new sheet to their account (however the account model functionality is still in progress). For now a good test is to just to post the endpoint with your credentials from the JSON you were using previously as well as the URL and then in the response, bet back the list of applicants.

### PR DESCRIPTION
**Pull request by kavpreetgrewal**
Currently just using the API credentials from the JSON, but can be switched over once the OAuth is all set up.
Closes #29

### CODE REVIEW

File: src/controllers/sheets.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { getSheetData } from '../utils/gsuite/sheets';
+import { Applicant } from '../types/applicant';
+
+/**
+ * @description Gets the applicants using the Google Sheet info from the given JSON object in request body
+ *              And sends the array of objects
+ */
+export const postSheet = async (req, res) => {
```
Comment (DConnorFong) [inline src/controllers/sheets.ts]: Warning for return type on function

**Review by DConnorFong — CHANGES_REQUESTED** [review]
Adding a general design suggestion - I think we should create a new model for a Sheet. It would contain the Sheet ID as well as the owner's email (replace with ID later).

This has the advantage that we can protect permissions to delete a sheet, and prevent duplicate additions.

Whenever we POST a new sheet, it would just check the sheet model to see if the ID exists, and if it does then just abort and return a 400.

**Comment by kavpreetgrewal** [comment]
> Whenever we POST a new sheet, it would just check the sheet model to see if the ID exists, and if it does then just abort and return a 400

@DConnorFong Would we then also have to check if the sheet has been edited? I guess it will work out better once we connect this with the database and then just perform checks on each applicant before adding.

**Comment by DConnorFong** [comment]
> > Whenever we POST a new sheet, it would just check the sheet model to see if the ID exists, and if it does then just abort and return a 400
> 
> @DConnorFong Would we then also have to check if the sheet has been edited? I guess it will work out better once we connect this with the database and then just perform checks on each applicant before adding.

@KavpreetGrewal that's a pretty good point - its probably not a super important factor unless you wanted to split the logic into a function that assumes there are no applicants and a function that assumes there are some. The deleting is probably most important to enforce who has permissions to do certain authorities.

**Review by DConnorFong — APPROVED** [review]
LGTM

**Pull request merged**